### PR TITLE
Create service updates index

### LINKED ISSUE #429: AUTOMATIC BATCH PROCESS - Create service inserts new file into index
### User want
As a user
I want to see up to date documents on the products website
So I can make informed decisions

**Customer acceptance criteria** 

**Technical acceptance criteria**
Create service inserts metadata for the newly uploaded file into the search index. I don't think there's currently a Rust client for this but it should be achievable through the Azure API using a HTTP client.

**Data acceptance criteria**

**Testing acceptance criteria** 

**Size**

**Value**

**Effort**

### Exit Criteria met 
- [x] Backlog
- [x] Discovery
- [x] DUXD
- [ ] Development 
- [ ] Quality Assurance 
- [ ] Release and Validate

### PR DESCRIPTION
**Pull request by rjkerrison**
# Create service updates index

Closes #429. Continues from PR #506. Replaces PR #517.

Adding a Lapras to a Pokédex

### Acceptance Criteria

- [ ] After blobs are created, the index should be updated to reflect the new blob

### Testing information

Run `make`.

### Pre-flight Checklist

- [ ] Testing aligns with [testing strategy][testing strategy]
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/doc-index-updater/src/create_manager/models.rs [hunk 1]
```diff
@@ -128,12 +223,12 @@ mod test {
         let expected_doc_type = "Spc".to_string();
         let expected_title = "Paracetamol Plus PL 12345/6789".to_string();
         let expected_author = "JRR Tolkien".to_string();
-        let expected_product_name = "[\"Effective product 1\",\"Effective product 2\"]".to_string();
-        let expected_substance_name = "[\"Paracetamol\",\"Caffeine\"]".to_string();
+        let expected_product_name = "[\"EFFECTIVE PRODUCT 1\",\"EFFECTIVE PRODUCT 2\"]".to_string();
+        let expected_substance_name = "[\"PARACETAMOL\",\"CAFFEINE\"]".to_string();
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/models.rs]: Just to double-check—this case conversion is something we're expected to do to normalise unpredictable data, yes?

File: medicines/doc-index-updater/src/create_manager/models.rs [hunk 2]
```diff
@@ -1,43 +1,138 @@
...
+            content: "Content not yet available".to_owned(),
+            rev_label: "1".to_owned(),
+            product_name: blob.product_names.join(", "),
+            created: Utc::now().to_rfc3339_opts(SecondsFormat::Secs, true),
+            release_state: "Y".to_owned(),
+            keywords: blob.keywords.to_owned().unwrap_or_default().join(", "),
+            title: blob.title.to_owned(),
+            pl_number: vec![blob.pl_number.to_owned()],
+            file_name: blob.file_name.to_owned(),
+            doc_type: blob.doc_type.to_string(),
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/models.rs]: Why the mix of `.to_owned()` and `.to_string()` here? Aren't they the same thing?

File: medicines/doc-index-updater/src/create_manager/search_index.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+use crate::{
+    create_manager::models::{BlobMetadata, IndexEntry},
+    search_client,
+};
+
+//use std::time::{Duration, Instant};
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/search_index.rs]: Should be removed.

File: medicines/doc-index-updater/src/create_manager/search_index.rs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    file_size: usize,
+) -> Result<(), anyhow::Error> {
+    let storage_account =
+        std::env::var("STORAGE_ACCOUNT").expect("Set env variable STORAGE_ACCOUNT first!");
+    let container_name =
+        std::env::var("STORAGE_CONTAINER").expect("Set env variable STORAGE_CONTAINER first!");
+    let storage_path = format!(
+        "https://{}.blob.core.windows.net/{}/{}",
+        &storage_account, &container_name, &blob_name
+    );
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/search_index.rs]: This is fine, but is there a way to get `create_blob` to return this URL along with the blob name?

File: medicines/doc-index-updater/src/create_manager/models.rs (hunk 1 above)
Comment (rjkerrison) [reply medicines/doc-index-updater/src/create_manager/models.rs]: yeah, I'm doing it here so BLOB matches INDEX and the DRUG INDEX/SUBSTANCE pages work as they have done previously
@TimboTambo and @m-doughty as importer experts - does this seem right to you?

File: medicines/doc-index-updater/src/create_manager/models.rs (hunk 2 above)
Comment (rjkerrison) [reply medicines/doc-index-updater/src/create_manager/models.rs]: Absolutely not. `doc_type` is a `DocumentType`, so I need to convert it to a `String`. The others are `String` properties of the `blob`, and I need to let the new `IndexEntry` own a copy of them, so I say `to_owned()`.

File: medicines/doc-index-updater/src/create_manager/search_index.rs (hunk 4 above)
Comment (rjkerrison) [reply medicines/doc-index-updater/src/create_manager/search_index.rs]: Great idea, I've gone pretty far with it. There's now a `Blob` which has everything we need for adding to the index.

File: medicines/doc-index-updater/src/create_manager/models.rs (hunk 1 above)
Comment (m-doughty) [reply medicines/doc-index-updater/src/create_manager/models.rs]: Seems right to me.

**Pull request merged**